src/layers/fp64/arc-layer/arc-layer-64.js

```diff
@@ -18,104 +18,51 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-import {Layer} from '../../../lib';
-import {assembleShaders} from '../../../shader-utils';
+import ArcLayer from '../../core/arc-layer';
 import {fp64ify} from '../../../lib/utils/fp64';
-import {GL, Model, Geometry} from 'luma.gl';
 import {readFileSync} from 'fs';
 import {join} from 'path';
 
-const DEFAULT_COLOR = [0, 0, 255, 255];
-
-const defaultGetSourcePosition = x => x.sourcePosition;
-const defaultGetTargetPosition = x => x.targetPosition;
-const defaultGetColor = x => x.color;
-
-const defaultProps = {
-  getSourcePosition: defaultGetSourcePosition,
-  getTargetPosition: defaultGetTargetPosition,
-  getSourceColor: defaultGetColor,
-  getTargetColor: defaultGetColor,
-  strokeWidth: 1
-};
-
-export default class ArcLayer64 extends Layer {
-  constructor(props) {
-    super(Object.assign({}, defaultProps, props));
-  }
+export default class ArcLayer64 extends ArcLayer {
 
   initializeState() {
-    const {gl} = this.context;
+    super.initializeState();
+
     const {attributeManager} = this.state;
 
     attributeManager.addInstanced({
-      instanceSourcePositionsFP64: {
-        size: 4,
-        update: this.calculateInstanceSourcePositions
-      },
-      instanceTargetPositionsFP64: {
-        size: 4,
-        update: this.calculateInstanceTargetPositions
-      },
-      instanceSourceColors: {
+      instanceSourcePositions64: {
         size: 4,
-        type: GL.UNSIGNED_BYTE,
-        update: this.calculateInstanceSourceColors
+        update: this.calculateInstanceSourcePositions64
       },
-      instanceTargetColors: {
+      instanceTargetPositions64: {
         size: 4,
-        type: GL.UNSIGNED_BYTE,
-        update: this.calculateInstanceTargetColors
+        update: this.calculateInstanceTargetPositions64
       }
+      // Reuse from base class
+      // instanceSourceColors: {
```

Comment:
Maybe just mentioning the attributes we reuse is enough - i.e. a one line comment